ts_src/psetv2/zkp.ts

```diff
@@ -487,42 +480,47 @@ export class ZKPGenerator {
     throw new Error('Could not unblind output with any blinding key');
   }
 
-  private getInputAssetsAndBlinders(
-    pset: Pset,
-    issuanceBlindingArgs?: IssuanceBlindingArgs[],
-  ): {
+  private getInputAssetsAndBlinders(pset: Pset): {
     assets: Buffer[];
     assetBlinders: Buffer[];
   } {
+    const assets: Buffer[] = [];
+    const assetBlinders: Buffer[] = [];
+
     const unblindedIns = this.maybeUnblindInUtxos(pset);
+
+    for (const unblindedIn of unblindedIns) {
+      assets.push(unblindedIn.asset);
+      assetBlinders.push(unblindedIn.assetBlindingFactor);
+    }
+
     pset.inputs.forEach((input, i) => {
       if (input.hasIssuance() || input.hasReissuance()) {
-        unblindedIns.push({
-          value: '',
-          valueBlindingFactor: Buffer.from([]),
-          asset: input.getIssuanceAssetHash()!,
-          assetBlindingFactor: ZERO,
-        });
-        if (input.issuanceInflationKeys! > 0) {
-          const isBlindedIssuance =
-            issuanceBlindingArgs! &&
-            issuanceBlindingArgs.find(({ index }) => index === i) !== undefined;
-          unblindedIns.push({
-            value: '',
-            valueBlindingFactor: Buffer.from([]),
-            asset: input.getIssuanceInflationKeysHash(isBlindedIssuance)!,
-            assetBlindingFactor: ZERO,
-          });
-        }
-      }
-    });
+        const issAssetHash = input.getIssuanceAssetHash();
+        if (!issAssetHash)
+          throw new Error(
+            `something went wrong while getting the issuance asset hash on input #${i}`,
+          );
+
+        assets.push(issAssetHash);
+        assetBlinders.push(ZERO);
 
-    const assets = [] as Buffer[];
-    const assetBlinders = [] as Buffer[];
+        if (!input.hasReissuance() && input.issuanceInflationKeys! > 0) {
+          const blindedIssuance = input.blindedIssuance;
+          if (blindedIssuance === undefined)
+            throw new Error(`input #${i} is missing blindedIssuance field`);
 
-    unblindedIns.forEach(({ asset, assetBlindingFactor }) => {
-      assets.push(asset);
-      assetBlinders.push(assetBlindingFactor);
+          const inflationTokenAssetHash =
+            input.getIssuanceInflationKeysHash(blindedIssuance);
```

Comment:
this method doesn't need the bool arg anymore since it's already contained by the input itself.